mock/mock_test.go

```diff
@@ -937,6 +937,30 @@ func Test_Mock_Return_NotBefore_In_Order(t *testing.T) {
 	})
 }
 
+func Test_Mock_Return_InOrder_Uses_NotBefore(t *testing.T) {
+	var mockedService = new(TestExampleImplementation)
+
+	b := mockedService.
+		On("TheExampleMethod", 1, 2, 3).
+		Return(4, nil)
+	c := mockedService.
+		On("TheExampleMethod2", true).
+		Return()
+
+	InOrder(
+		b,
+		c,
+	)
+
+	require.Equal(t, []*Call{b, c}, mockedService.ExpectedCalls)
```

Comment:
This line is testing the implementation. A cost effective test would only test the behaviour, as you are doing just fine after this line.

If you remove this assertion then we are free to change the implementation of the intended behaviour without breaking the test. It also removes the need to declare `b` and `c`, so the test can be written as `InOrder` is intended to be used.

Reply:
Also, it's just comparing pointer values. If InOrder did literally nothing then that assertion would still pass.